test_cluster_flush_slots_after_config_change

https://github.com/dragonflydb/dragonfly/actions/runs/24329801324/job/71032813968

```
>       assert await c_replica.execute_command("dbsize") == (100_000 - slot_0_size)
E       assert equals failed
E         99998  99993

c_master   = <redis.asyncio.client.Redis(<redis.asyncio.connection.ConnectionPool(<redis.asyncio.connection.Connection(db=0,usernam...tocol=2,host=localhost,port=30017,socket_connect_timeout=None,socket_keepalive=None,socket_keepalive_options=None)>)>)>
c_master_admin = <redis.asyncio.client.Redis(<redis.asyncio.connection.ConnectionPool(<redis.asyncio.connection.Connection(db=0,usernam...tocol=2,host=localhost,port=30018,socket_connect_timeout=None,socket_keepalive=None,socket_keepalive_options=None)>)>)>
c_replica  = <redis.asyncio.client.Redis(<redis.asyncio.connection.ConnectionPool(<redis.asyncio.connection.Connection(db=0,usernam...tocol=2,host=localhost,port=30019,socket_connect_timeout=None,socket_keepalive=None,socket_keepalive_options=None)>)>)>
c_replica_admin = <redis.asyncio.client.Redis(<redis.asyncio.connection.ConnectionPool(<redis.asyncio.connection.Connection(db=0,usernam...tocol=2,host=localhost,port=30020,socket_connect_timeout=None,socket_keepalive=None,socket_keepalive_options=None)>)>)>
config     = '\n      [\n        {\n          "slot_ranges": [\n            {\n              "start": 1,\n              "end": 1638...",\n              "ip": "localhost",\n              "port": 9001\n            }\n          ]\n        }\n      ]\n    '
df_factory = Factory({'proactor_threads': 4, 'cluster_mode': 'yes'})
master     = :30017
master_id  = '92302a9b5d2c4dcc06d8bc75a2e711203fcff8a4'
replica    = :30019
replica_id = 'fe870d8eb019b0d5c3f31e0294ec0da0df5ddf35'
resp       = [[0, 'key_count', 7, 'total_reads', 0, 'total_writes', ...]]
slot_0_size = 7

../tests/dragonfly/cluster_test.py:958: AssertionError
```

--- a/tests/dragonfly/cluster_test.py
+++ b/tests/dragonfly/cluster_test.py
@@ -1,23 +1,17 @@
-import pytest
 import copy
-import re
-import json
-import redis
 from binascii import crc_hqx
-from redis import asyncio as aioredis
-import asyncio
 from dataclasses import dataclass
 
-from .instance import DflyInstanceFactory, DflyInstance
-from .utility import *
-from .replication_test import check_all_replicas_finished
-from redis.cluster import RedisCluster
 from redis.cluster import ClusterNode
+from redis.cluster import RedisCluster
 from redis.exceptions import MovedError
-from .proxy import Proxy
-from .seeder import Seeder, SeederBase, DebugPopulateSeeder
 
 from . import dfly_args
+from .instance import DflyInstanceFactory, DflyInstance
+from .proxy import Proxy
+from .replication_test import check_all_replicas_finished
+from .seeder import DebugPopulateSeeder
+from .utility import *
 
 BASE_PORT = 30001
 
@@ -953,9 +947,22 @@ async def test_cluster_flush_slots_after_config_change(df_factory: DflyInstanceF
     await push_config(config, [c_master_admin, c_replica_admin])
 
     await check_all_replicas_finished([c_replica], c_master)
+    expected_size = 100_000 - slot_0_size
+    cmd = ("DFLYCLUSTER", "GETSLOTINFO", "SLOTS", "0")
 
-    assert await c_master.execute_command("dbsize") == (100_000 - slot_0_size)
-    assert await c_replica.execute_command("dbsize") == (100_000 - slot_0_size)
+    @assert_eventually(timeout=5)
+    async def slot_zero_flushed():
+        m_slots, repl_slots, master_size, replica_size = await asyncio.gather(
+            c_master_admin.execute_command(*cmd),
+            c_replica_admin.execute_command(*cmd),
+            c_master.execute_command("DBSIZE"),
+            c_replica.execute_command("DBSIZE"),
+        )
+
+        assert m_slots[0][2] == repl_slots[0][2] == 0
+        assert master_size == replica_size == expected_size
+
+    await slot_zero_flushed()
 
 
 @dfly_args({"proactor_threads": 4, "cluster_mode": "yes", "admin_port": next(next_port)})

